The dependency on keras(-nightly) seems to be superflous.
See https://github.com/tensorflow/tensorflow/issues/49298

Author: Alexander Grund (TU Dresden)

diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index 354657a7bfc..335f1f638ad 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -102,9 +102,6 @@ REQUIRED_PACKAGES = [
     # When updating these, please also update the nightly versions below
     'tensorboard ~= 2.5',
     'tensorflow-estimator >= 2.5.0rc0 , < 2.6.0',
-    # TODO(scottzhu): OSS keras hasn't been formally released yet.
-    # Use keras-nightly at the moment.
-    'keras-nightly ~= 2.5.0.dev',
 ]
 
 
